forge-game/src/main/java/forge/game/ability/effects/RepeatEachEffect.java

```diff
@@ -139,6 +137,36 @@ else if (sa.hasParam("DefinedCards")) {
             }
         }
 
+        if (sa.hasParam("RepeatTypes")) {
+            final String repTypes = sa.getParam("RepeatTypes");
+            final List<String> validTypes = new ArrayList<>();
+            if (repTypes.startsWith("ThisTurnCast")) {
+                final String[] workingCopy = repTypes.split("_");
+                final String validFilter = workingCopy[1];
+                List<Card> res = CardUtil.getThisTurnCast(validFilter, source, sa);
+                for (final Card c : res) {
+                    for (String t : c.getType()) {
```

Comment:
Not sure if adding all subtypes will be worth it, otherwise you could just use `getCardTypesFromList`

Reply:
Does this also grab subtypes? I thought it only grabbed core types 😬